test/metabase/api/session_test.clj

```diff
@@ -569,6 +569,11 @@
 
 ;;; ------------------------------------------- TESTS FOR UNSUBSCRIBING NONUSERS STUFF --------------------------------------------
 
+(deftest unsubscribe-hash-test
+  (with-redefs [public-settings/site-uuid-for-unsubscribing-url (constantly "08534993-94c6-4bac-a1ad-86c9668ee8f5")]
+    (is (= "691501b28502126a6f59636ce147d26afa9d8fe8e5d0bd41f0de609194c0d9ba0042276d9b23cf10b1c7a4b18b3f946707e511d3e95e08fbe2c456080f619e36"
+           (messages/generate-pulse-unsubscribe-hash :pulse-id "rasta@pasta.com")))))
```

Comment:
This tests that the hash is stable, but doesn't test the security-related property that the `generate-pulse-unsubscribe-hash` return value depends on `site-uuid-for-unsubscribing-url`. What do you think about adding a test for that?

Reply:
Added, along with rebasing on master so we can stub out read-only settings more idiomatically.